Essentials/Commands/BlocksModule.cs

```diff
@@ -140,122 +140,175 @@ public void OffSubtype(string subtype)
+                                f.FatBlock.BlockDefinition.TypeId == typeof(MyObjectBuilder_BatteryBlock) ||
+                                f.FatBlock.BlockDefinition.TypeId == typeof(MyObjectBuilder_SolarPanel)));
+                                var list = blocks.Select(f => (IMyFunctionalBlock)f.FatBlock).Where(f => !f.Enabled).ToArray();
+                                foreach (var item in list)
+                                {
+                                    item.Enabled = true;
+                                }
+                                count++;
+                            }
+                        }
+                        break;
+                    case blockcategory.production:
+                        {
+                            foreach (var entity in MyEntities.GetEntities().OfType<MyCubeGrid>())
+                            {
+                                IMyCubeGrid grid = entity as MyCubeGrid;
+                                var blocks = new List<IMySlimBlock>();
+                                grid.GetBlocks(blocks, f => f.FatBlock != null && f.FatBlock is IMyFunctionalBlock
                     && (f.FatBlock.BlockDefinition.TypeId == typeof(MyObjectBuilder_Refinery) ||
                     f.FatBlock.BlockDefinition.TypeId == typeof(MyObjectBuilder_Assembler) ||
                     f.FatBlock.BlockDefinition.TypeId == typeof(MyObjectBuilder_OxygenGenerator)));
-                    var list = blocks.Select(f => (IMyFunctionalBlock)f.FatBlock).Where(f => !f.Enabled).ToArray();
-                    foreach (var item in list)
-                    {
-                        item.Enabled = true;
-                    }
-                    count++;
-
-                }
-                if (general.Contains("wea"))
-                {
-                    status = "Weapon";
-                    var blocks = new List<IMySlimBlock>();
-                    grid.GetBlocks(blocks, f => f.FatBlock != null && f.FatBlock is IMyFunctionalBlock
+                                var list = blocks.Select(f => (IMyFunctionalBlock)f.FatBlock).Where(f => !f.Enabled).ToArray();
+                                foreach (var item in list)
+                                {
+                                    item.Enabled = true;
+                                }
+                                count++;
+                            }
+                        }
+                        break;
+                    case blockcategory.weapons:
+                        {
+                            foreach (var entity in MyEntities.GetEntities().OfType<MyCubeGrid>())
+                            {
+                                IMyCubeGrid grid = entity as MyCubeGrid;
+                                var blocks = new List<IMySlimBlock>();
+                                grid.GetBlocks(blocks, f => f.FatBlock != null && f.FatBlock is IMyFunctionalBlock
```

Comment:
This lambda is bad, and you have this enormous type check duplicated in many places. You should move the check into a new method: `private bool IsBlockOfType(MySlimBlock block, blockcategory category)`

Reply:
Good Idea.  Thanks